client/components/common/queries.js

```diff
@@ -0,0 +1,183 @@
+import { gql } from '@apollo/client';
```

Comment:
I'm down with this. Most fields don't impact query performance so it's totally fine to just query all of them every time. For fields that do impact performance, like tests on TestPlanVersions and testResults on TestPlanRuns, I think it should be divided into two fragments to make accidentally tanking your performance less likely to happen. Something like `TEST_PLAN_RUN_FIELDS` and `TEST_PLAN_RUN_TEST_RESULTS_FIELDS`.

Reply:
I think fields like renderableContent might be preferable to leave off, since they're extremely verbose and "specialty" for specific use cases. If it's there where it's not needed it will definitely make debugging responses harder.

Reply:
I can see that benefit, as well as some others, especially in the mock testing area and such. For clarity though, the fields replaced by a fragment has the exact same fields as were listed before, so no additional _(or unnecessary)_ fields just yet. But I do think there's a benefit to this approach, and an even greater benefit if that structure is set up **properly**.

So I'd rather to remove this proposed enhancement bit from these unrelated code changes and follow up in a separate issue so this enhancement can get it's own focused space.